x-pack/plugins/apm/public/components/shared/unified_search_bar/index.tsx

```diff
@@ -0,0 +1,182 @@
+  showSubmitButton = true,
+  isClearable = true,
+}: {
+  placeholder?: string;
+  value?: string;
+  showDatePicker?: boolean;
+  showSubmitButton?: boolean;
+  isClearable?: boolean;
+}) {
+  const {
+    unifiedSearch: {
+      ui: { SearchBar },
+    },
+    core,
+  } = useApmPluginContext();
+  const { services } = useKibana<ApmPluginStartDeps>();
+
+  const {
+    data: {
+      query: { queryString: queryStringService, timefilter: timeFilterService },
+    },
+  } = services;
+
+  const urlQuery = useKueryParams(value);
+  const timePickerTimeDefaults = core.uiSettings.get<TimePickerTimeDefaults>(
+    UI_SETTINGS.TIMEPICKER_TIME_DEFAULTS
+  );
+  const urlTimeRange = useUrlTimeRange(timePickerTimeDefaults);
+  const [displaySearchBar, setDisplaySearchBar] = useState(false);
+
+  const syncSearchBarWithUrl = useCallback(() => {
+    // Sync Kuery params with Search Bar
+    if (urlQuery && !deepEqual(queryStringService.getQuery(), urlQuery)) {
+      queryStringService.setQuery(urlQuery);
+    }
+    // On page navigation the search bar persists the state where as the url is cleared, hence we need to clear the search bar
+    if (!urlQuery) {
+      queryStringService.clearQuery();
+    }
+    // Sync Time Range with Search Bar
+    timeFilterService.timefilter.setTime(urlTimeRange as TimeRange);
+  }, [urlQuery, queryStringService, timeFilterService, urlTimeRange]);
+
+  useEffect(() => {
+    syncSearchBarWithUrl();
+  }, [syncSearchBarWithUrl]);
+
+  const location = useLocation();
+  const history = useHistory();
+  const { dataView } = useApmDataView();
+  const searchbarPlaceholder = useSearchBarPlaceholder(placeholder);
+
+  useEffect(() => {
+    if (dataView) setDisplaySearchBar(true);
+  }, [dataView]);
+  const handleSubmit = (payload: { dateRange: TimeRange; query?: Query }) => {
+    const { dateRange, query } = payload;
+    const { from: rangeFrom, to: rangeTo } = dateRange;
+    const toQueryId = toQuery(location.search);
+    const updatedQueryWithTime = { ...toQueryId, rangeFrom, rangeTo };
+    const fromQueryId = fromQuery({
+      ...updatedQueryWithTime,
+      kuery: query?.query,
+    });
+    const queryToPush = {
+      ...location,
+      search: fromQueryId,
+    };
+    history.push(queryToPush);
+  };
+
+  return (
+    <EuiSkeletonRectangle
+      isLoading={!displaySearchBar}
+      width="100%"
+      height="40px"
+    >
+      <SearchBar
+        appName={i18n.translate('xpack.apm.appName', {
+          defaultMessage: 'Service Transaction',
+        })}
+        iconType="search"
+        placeholder={searchbarPlaceholder}
+        useDefaultBehaviors={true}
+        indexPatterns={dataView ? [dataView] : undefined}
+        showQueryInput={true}
+        showQueryMenu={false}
+        showFilterBar={false}
+        showDatePicker={showDatePicker}
+        showSubmitButton={showSubmitButton}
+        displayStyle="inPage"
+        onQuerySubmit={handleSubmit}
+        isClearable={isClearable}
```

Comment:
in case `isClearable` is true it seems there's no handler for that (ex `onClear or onReset`) to refresh the table/view. Is this an expected behaviour to ship?

Reply:
Unified Search does not expose any such handler to handle `onClear/onReset`. For them everything is either by pressing enter on the Input Bar or Clicking the Submit button.

The reason i exposed this `isClearable ` intentionally was, there is a page, not sure Storage Explorer or Agent Explorer where we only display the Input Bar and there is no DateTimePicker and no Submit button. Imagine if they had this `Clear Crossmark` and they click on it. Then they would expect some sort of `onClear` or `onReset` to be triggered automatically to refresh the page. But since we Unified Search does not exposes such handlers, we cannot do that. So to force the user to press `enter`, i removed the `clear crossmark` on that page. This means, they will have to use the Keyboard to clear the text and press enter.

Reply:
@kpatticha I just realised what you meant to say. Yes this is an expected behaviour. The Unified Searchbar turns changes the button text and color to inform user that an action is required.


https://user-images.githubusercontent.com/7416358/231421274-a630f67c-e7ec-4a61-9c45-ffa6e7914418.mov

Reply:
@achyutjhunjhunwala thanks.